pkg/scheduler/cache/status_updater/concurrency.go

```diff
@@ -30,24 +30,53 @@ func (su *defaultStatusUpdater) SyncPodGroupsWithPendingUpdates(podGroups []*eng
 	for i := range podGroups {
 		key := su.keyForPodGroupPayload(podGroups[i].Name, podGroups[i].Namespace, podGroups[i].UID)
 		usedKeys[key] = true
-		inflightUpdateAny, found := su.inFlightPodGroups.Load(key)
-		if !found {
+		pgLatestUpdate, inFlightUpdateFound, appliedUpdateFound := su.getLatestPgUpdate(key)
+		if !inFlightUpdateFound && !appliedUpdateFound {
 			continue
 		}
-		podGroup := inflightUpdateAny.(*inflightUpdate).object.(*enginev2alpha2.PodGroup)
-		isPodGroupUpdated := su.syncPodGroup(podGroup, podGroups[i])
-		if isPodGroupUpdated {
-			su.inFlightPodGroups.Delete(key)
+		podGroup := pgLatestUpdate.object.(*enginev2alpha2.PodGroup)
+		podGroupMatchesUpdate := su.syncPodGroup(podGroup, podGroups[i])
+		// Delete the inflight update if it was applied + the pod group in the lister matches the inFlight
+		if podGroupMatchesUpdate {
+			su.cleanPgUpdatesCaching(key, appliedUpdateFound)
```

Comment:
```suggestion
		if podGroupMatchesUpdate && appliedUpdateFound {
			su.appliedPodGroupUpdates.Delete(key)
```

and then you can delete the function `cleanPgUpdatesCaching`